Warning building sha code with mbedtls

An API in mbedtls has changed in v3 which we're now using. If you build the pico-examples sha example it shows this...

```
/pico-sdk-dev/src/rp2_common/pico_mbedtls/pico_mbedtls.c:58:71: warning: argument 2 of type 'unsigned char[32]' with mismatched bound [-Warray-parameter=]
   58 | int mbedtls_sha256_finish( mbedtls_sha256_context *ctx, unsigned char output[32]) {
      |                                                         ~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /pico-sdk-dev/src/rp2_common/pico_mbedtls/pico_mbedtls.c:10:
/pico-sdk-dev/lib/mbedtls/include/mbedtls/sha256.h:128:42: note: previously declared as 'unsigned char *'
  128 |                           unsigned char *output);
      |                           ~~~~~~~~~~~~~~~^~~~~~
[  0%] Built target hello_sha256
```

--- a/src/rp2_common/pico_mbedtls/pico_mbedtls.c
+++ b/src/rp2_common/pico_mbedtls/pico_mbedtls.c
@@ -55,7 +55,11 @@ int mbedtls_sha256_update(mbedtls_sha256_context *ctx, const unsigned char *inpu
     return 0;
 }
 
+#if MBEDTLS_VERSION_MAJOR < 3
 int mbedtls_sha256_finish( mbedtls_sha256_context *ctx, unsigned char output[32]) {
+#else
+int mbedtls_sha256_finish( mbedtls_sha256_context *ctx, unsigned char *output) {
+#endif
     sha256_result_t result;
     pico_sha256_finish(ctx, &result);
     memcpy(output, result.bytes, 32);

